packages/vkui/src/components/Button/Button.tsx

```diff
@@ -99,25 +116,69 @@ export const Button = ({
   after,
   getRootRef,
   loading,
+  loadingLabel = 'Загрузка...',
   onClick,
   disableSpinnerAnimation,
   rounded,
   disabled,
+  href,
+  'aria-label': ariaLabelProp,
   ...restProps
 }: ButtonProps): React.ReactNode => {
   const hasIconOnly = !children && Boolean(after) !== Boolean(before);
   const { sizeY = 'none' } = useAdaptivity();
   const platform = usePlatform();
 
+  const isDisabled = disabled || loading;
+  const hasHref = href !== undefined;
+
+  const ariaLabel = loading ? loadingLabel : ariaLabelProp;
+
+  const handleClick = React.useCallback(
+    (e: React.MouseEvent<HTMLElement>) => {
+      if (isDisabled) {
+        e.preventDefault();
+        return;
+      }
+      onClick?.(e);
+    },
+    [isDisabled, onClick],
+  );
+
+  const buttonProps = React.useMemo(() => {
+    if (hasHref) {
+      return isDisabled
+        ? {
+            // Для disabled/loading ссылок нужно удалить href и добавить role="link"
+            // согласно https://w3c.github.io/html-aria/#example-communicate-a-disabled-link-with-aria
+            'role': 'link' as const,
+            'Component': 'a' as const,
+            'aria-disabled': isDisabled,
+          }
+        : {
+            href,
+            'aria-disabled': isDisabled,
+          };
+    } else {
+      return {
+        Component: 'button' as const,
+        disabled,
+      };
+    }
+  }, [disabled, hasHref, href, isDisabled]);
+
   return (
     <Tappable
       hoverMode={styles.hover}
       activeMode={styles.active}
-      {...(restProps.href === undefined && { Component: 'button' })}
       focusVisibleMode="outside"
-      disabled={loading || disabled}
+      aria-busy={loading}
+      hasHover={!loading}
+      hasActive={!loading}
+      {...buttonProps}
       {...restProps}
-      onClick={loading ? undefined : onClick}
+      aria-label={ariaLabel}
+      onClick={handleClick}
```

Comment:
```suggestion
      onClick={isDisabled ? undefined : onClick}
```
а если так сделать, не вариант?

Reply:
не вижу смысла в `preventDefault()`

- при `disabled` в целом кнопка недоступна и клик не будет вызван никогда
- при `loading` поведение должно быть таким же как при `disabled`
- пользовательское событие в обеих вариантах не вызывается

поэтому проще и выгодней будет не создавать HOC над `onClick`